src/robot.cpp

```diff
@@ -408,6 +522,62 @@ void Robot::execute()
   }
 }
 
+geometry_msgs::Pose Robot::getManipulationTarget()
+{
+  std::string planning_group_name = config_->getFeatureManipulation().getActivePlanningGroup();
+  
+  auto group_it = planning_groups_.find(planning_group_name);
+  TEMOTO_INFO(planning_group_name.c_str());
+
+  geometry_msgs::Pose current_pose;
+  
+  if (group_it != planning_groups_.end())
+  {    
+    current_pose = group_it->second->getCurrentPose().pose;    
+    //group_it->second->setNamedTarget("test_pose");
+  }
+  else 
+  {
+    //TODO: This section has to utilize temoto error management system
+    TEMOTO_ERROR("Planning group '%s' was not found.", planning_group_name.c_str());
+  } 
+  return current_pose;  
+}
+
+void Robot::goalNavigation(const std::string& planning_group_name, const geometry_msgs::PoseStamped& target_pose)
+{
+    FeatureNavigation& ftr = config_->getFeatureNavigation();
+    std::string act_rob_ns = config_->getAbsRobotNamespace() + "/move_base";
+    
+    MoveBaseClient ac(act_rob_ns, true);    
+    
+    while(!ac.waitForServer(ros::Duration(5.0)))
```

Comment:
What if the server does not come up, is the program going to be stuck here forever?